### PR TITLE
Add example of how to structure binary-data forms without data loss

### PR DESCRIPTION
**Pull request by gbourant**
Improve documentation for form submission with binary data.

### CODE REVIEW

File: www/content/docs.md [hunk 1]
```diff
@@ -618,6 +618,35 @@ differently.
 Note that htmx fires a `htmx:xhr:progress` event periodically based on the standard `progress` event during upload,
 which you can hook into to show the progress of the upload.
 
+##### File Upload and Binary Files {#files}
+
+When using server-side error handling with forms that include both primitive values and binary files, the binary file data is lost during the form's return with error messages. Consequently, users are required to re-upload the binary file, resulting in a less user-friendly experience.
+
+To overcome the problem of losing binary file data during form error handling, you can adopt the following approach:
```
Comment (Renerick) [inline www/content/docs.md:625]:
```suggestion
##### File Inputs {#files}

When using server-side error handling and validation with forms that include both primitive values and file inputs, the file input's value is lost when the form returns with error messages. Consequently, users are required to re-upload the file, resulting in a less user-friendly experience.

To overcome the problem of losing file input value in simple cases, you can adopt the following approach:
```

**Review by Renerick — COMMENTED** [review]
You make an excellent point with this section and I vote for inclusion of it to the docs. My only suggestion is to replace the words "binary file" with "file inputs" to more accurately describe the problem at hand. I've taken the liberty to submit code suggestion with the example how I personally would phrase it in the first paragraphs

**Comment by gbourant** [comment]
Thanks for your valuable feedback! I appreciate your help and have accepted the changes.

**Comment by alexpetros** [comment]
I think is a good thing to put on the website, but I think it might be better suited to its own page in the `/examples` section, because it's kind of a niche how-to rather than an example of how a particular config or attribute works. Maybe "Forms With Binary Input"?

**Comment by alexpetros** [comment]
Sure, go for it.

File: www/content/docs.md [hunk 2]
```diff
@@ -618,6 +618,11 @@ differently.
 Note that htmx fires a `htmx:xhr:progress` event periodically based on the standard `progress` event during upload,
 which you can hook into to show the progress of the upload.
 
+<ul>
+    <li>
+    [How to preserve File Inputs after Server-Side Error Form Handling](@/examples/file-upload-input.md)</li>
+</ul>
+
```
Comment (alexpetros) [inline www/content/docs.md:625]: Can you change this to the following:

```
See the [examples section](@/examples/_index.md) for more advanced form patterns, including [progress bars](@/examples/file-upload.md) and [error handling](@/examples/file-upload-input.md).
```

File: www/content/examples/_index.md [hunk 3]
```diff
@@ -29,6 +29,7 @@ You can copy and paste them and then adjust them for your needs.
 | [Value Select](@/examples/value-select.md) | Demonstrates making the values of a select dependent on another select
 | [Animations](@/examples/animations.md) | Demonstrates various animation techniques
 | [File Upload](@/examples/file-upload.md) | Demonstrates how to upload a file via ajax with a progress bar
+| [File Upload And Server-Side Error Form Handling](@/examples/file-upload-input.md) | Demonstrates how to preserve File Inputs after Server-Side Error Form Handling
```
Comment (alexpetros) [inline www/content/examples/_index.md:32]: Please change this to:
```
[Preserving File Inputs after Form Errors](@/examples/file-upload-input.md) | Demonstrates how to preserve file inputs after form errors
```

File: www/content/examples/file-upload-input.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
++++
+title = "Preserving File Inputs after Server-Side Error Form Handling"
```
Comment (alexpetros) [inline www/content/examples/file-upload-input.md:2]: And this to:

```
Preserving File Inputs after Form Errors
```

**Review by alexpetros — COMMENTED** [review]
Just need some tiny changes to the titles and the links, and then I'll merge it. Thanks for the help!

**Review by alexpetros — APPROVED** [review]

**Pull request merged**